test/goose/brokers/redis/api_test.clj

```diff
@@ -201,3 +202,31 @@
              (-> (cron-jobs/find-by-name tu/redis-producer "my-cron-entry")
                  (:job-description)
                  (select-keys [:execute-fn-sym :args])))))))
+
+(deftest batch-test
+  (testing "[redis] batch API"
+    (let [batch-id (:id (goose.client/perform-batch tu/redis-client-opts
+                                                    {}
+                                                    `tu/my-fn
+                                                    (-> []
+                                                        (c/accumulate-batch-args "arg-one")
+                                                        (c/accumulate-batch-args "arg-two"))))
+          expected-batch {:id         batch-id
+                          :status     "in-progress"
+                          :total      2
+                          :enqueued   2
+                          :retrying   0
+                          :successful 0
+                          :dead       0}
+          {:keys [created-at] :as batch} (batch/find-by-id tu/redis-producer batch-id)]
+      (is (= expected-batch (dissoc batch :created-at)))
+      (is (Long/parseLong created-at))))
+  (testing "[redis] batch API with invalid id"
+    (let [invalid-id (str (random-uuid))
+          _ (goose.client/perform-batch tu/redis-client-opts
```

Comment:
Do we need to create a batch for this test?

Reply:
Creating a batch (using `perform-batch`) here so that the value returned by the api can be compared with it, to validate the correctness of the api fn.

Do you suggest a different approach to doing this?

Reply:
What you've suggested is required for the `[redis] batch API` test. For `[redis] batch API with invalid id`, we can avoid creating a batch and just assert nil? directly.

Reply:
Makes sense, removed the call to `perform-batch` here. And inlined `invalid-id`